presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -74,19 +75,30 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Sets.newConcurrentHashSet;
+import static java.lang.System.lineSeparator;
+import static java.util.Arrays.asList;
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.Executors.newCachedThreadPool;
 import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
 import static java.util.concurrent.TimeUnit.MICROSECONDS;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.joining;
+
 
 @ThreadSafe
 public class TaskExecutor
 {
     private static final Logger log = Logger.get(TaskExecutor.class);
 
     // print out split call stack if it has been running for a certain amount of time
-    private static final Duration LONG_SPLIT_WARNING_THRESHOLD = new Duration(600, TimeUnit.SECONDS);
+    private static final Duration LONG_SPLIT_WARNING_THRESHOLD = new Duration(600, SECONDS);
+    // Interrupt a split if it is running longer than this AND it's blocked on something known
+    private static final Duration LONG_SPLIT_INTERRUPT_THRESHOLD = new Duration(600, SECONDS);
```

Comment:
I wonder if it makes sense to make it configurable. Say instead of having a `boolean interruptRunawaySplits` config property, how about having something like `Duration interruptRunawaySplitsTimeout = new Duration(600, SECONDS)`, and setting it to `null` would effectively disable it.

Reply:
Simplifying config. I wasn't going to make the duration configurable. It's only configurable for the unit test and that isn't user visible. It's multi-threaded so coordinating the threads was harder than just setting the timeouts really low to get the test to run fast.

Reply:
Made it a configurable duration.